contracts/marketplace/src/test.rs

```diff
@@ -0,0 +1,229 @@
+#![cfg(test)]
+
+extern crate std;
+
+use crate::{MarketplaceContract, MarketplaceContractClient};
+use soroban_sdk::{
+    testutils::Address as _,
+    token::{self},
+    Address, Env,
+};
+
+fn setup_test<'a>() -> (Env, MarketplaceContractClient<'a>) {
+    let env = Env::default();
+    env.mock_all_auths();
+    let contract_id = env.register_contract(None, MarketplaceContract);
+    let client: MarketplaceContractClient<'_> = MarketplaceContractClient::new(&env, &contract_id);
+    (env, client)
+}
+
+fn create_token_asset<'a>(e: &Env, asset: &Address) -> token::StellarAssetClient<'a> {
+    token::StellarAssetClient::new(e, &e.register_stellar_asset_contract(asset.clone()))
+}
+
+#[test]
+#[should_panic(expected = "Error(Contract, #1)")]
+fn cannot_initialize_marketplace_twice() {
+    let (env, client) = setup_test();
+    let admin = Address::random(&env);
+    let asset = Address::random(&env);
+    let token = create_token_asset(&env, &asset);
+    client.init(&token.address, &admin);
+    client.init(&token.address, &admin);
+}
+
+#[test]
+#[should_panic(expected = "Error(Contract, #2)")]
+fn cannot_initialize_with_excess_percentage() {
+    let (env, client) = setup_test();
+    let admin = Address::random(&env);
+    let asset = Address::random(&env);
+    let token = create_token_asset(&env, &asset);
+    client.init(&token.address, &admin);
+}
+
+#[test]
+#[should_panic(expected = "Error(Contract, #2)")]
+fn cannot_initialize_with_negative_percentage() {
+    let (env, client) = setup_test();
+    let admin = Address::random(&env);
+    let asset = Address::random(&env);
+    let token = create_token_asset(&env, &asset);
+    client.init(&token.address, &admin);
+}
+
+#[test]
+fn can_create_listing() {
+    let (env, client) = setup_test();
+    let admin = Address::random(&env);
+    let seller = Address::random(&env);
+    let token = Address::random(&env);
+
+    let token = create_token_asset(&env, &token);
+    client.init(&token.address, &admin);
+
+    let asset = Address::random(&env);
```

Comment:
Should this `asset` be registered with `create_token_asset` instead of registering `token` ? On my mind `token` just represent the token with which we are going to trade things. Could this be part of the error we are getting [here](https://github.com/eigerco/nebula/pull/82#discussion_r1343736870) @geofmureithi ?

Reply:
This is referred to issue #90

Reply:
Ok, i understand we are going to look at this separatelly/inmediatelly after this merge right ? Because under my understanding, currently there is no real owner change more than in our contract internal state.